### PR TITLE
[MRG] Move tests to Github Actions

### PR DESCRIPTION
**Pull request by NicolasGensollen**
Move Travis tests to Github actions.
See #2627

### CODE REVIEW

**Comment by KamalakerDadi** [comment]
Can you rebase/merge on top of master? Or is it already the case? Because there were some PRs that were merged without Travis.

**Comment by NicolasGensollen** [comment]
>Can you rebase/merge on top of master? Or is it already the case? Because there were some PRs that were merged without Travis.

Done, thx!

I think this is ready for review. The actions test the following:

- Python `3.5` and minimum requirements on ubuntu
- Python `3.6`, `3.7`, and `3.8` on ubuntu, mac OS, and windows with latest package versions

Note: I had to modify (very slightly...) an assertion in a test to make it pass for Mac OS (see 0c8ed03).

**Comment by jeromedockes** [comment]
great, thanks! a couple of questions:
- shoudn't we remove `.travis.yml`?
- are the tests on windows in azure now redundant?

**Comment by NicolasGensollen** [comment]
>shoudn't we remove .travis.yml?

We could. I'm still waiting for the answer from the Travis Support people concerning my additional credit request. 
If they don't grant us enough credits, I was planning to use Travis only for testing on ARM architecture since this is a travis-specific feature for now.

>are the tests on windows in azure now redundant?

Yes (I dont think they run on the same version though, I'll have to double check...). I can remove windows tests from the github actions if this is too redundant.

**Comment by NicolasGensollen** [comment]
I just realized that the assertion that got broken on MacOS (see 0c8ed03) is also responsible for the test failure of #2608 
I'm not sure how the `150. * eps` threshold was chosen, and whether we can increase it up to `200. * eps` like I did here, but this might also solve #2608.

**Comment by bthirion** [comment]
Thx ! The changes look good.
I  think we should avoid redundancies btw azure and  Github-actions.

**Comment by bthirion** [comment]
There seems to be aproblem on some Python versions ?

**Comment by NicolasGensollen** [comment]
> There seems to be aproblem on some Python versions ?

Yes, I saw that yesterday. I think it was related to the very recent version bump of `importlib_metadata` from `3.1.1` to `3.3.0` which caused errors for Python versions `3.6` and `3.7` in the github actions workflows.  

The bug has been fixed, I re-ran all actions this morning and they all pass.

**Review by bthirion — APPROVED** [review]

**Comment by bthirion** [comment]
I'd be happy to get another review for that one.

**Comment by jeromedockes** [comment]
> I'd be happy to get another review for that one.
sure!

File: tools/github_actions_dependencies.sh
Comment (jeromedockes) [inline tools/github_actions_dependencies.sh:1]: why do we need '-f'?

File: tools/github_actions_install.sh
Comment (jeromedockes) [inline tools/github_actions_install.sh:1]: why '-f' ?

File: tools/github_actions_test.sh
Comment (jeromedockes) [inline tools/github_actions_test.sh:1]: why '-e' and '-f' ?
also we could use '-x' to avoid the need for the 'echo' -- and the printed command might get out of sync with what is actually executed below

File: tools/github_actions_install.sh
Comment (jeromedockes) [inline tools/github_actions_install.sh:4]: I wonder if we should have a job that checks that installation with pip runs smoothly?

File: tools/github_actions_test.sh
Comment (jeromedockes) [inline tools/github_actions_test.sh:4]: I think it would best not to run the tests from the directory in which the repo is checked out, but  from an empty (temporary) directory

**Comment by jeromedockes** [comment]
linking to the issue you opened on setup-python: https://github.com/actions/setup-python/issues/179

**Comment by NicolasGensollen** [comment]
@jeromedockes Thanks for the thorough review! :+1: 
Tests are now passing on MacOS (see the fix proposed in the issue you linked above). The only action that fails now is the `matplotlib-dev` tests which is allowed despite the disappointing red cross...
Feel free to let me know if something doesn't seem good enough for merging.

**Comment by jeromedockes** [comment]
great! indeed at the moment github actions doesn't seem to have an equivalent of `allow_failures`: https://github.com/actions/toolkit/issues/399
so I think for now I would remove the matplotlib-dev job (sorry I didn't realize this before, I'm new to github actions), and in a future PR we can see if we reintroduce it, either on github actions or the azure build, in a way that doesn't cause the overall build to fail.
I think my comment on running the tests in a separate directory is still relevant, but this PR is already a huge improvement over not running the tests at all on linux (which is the current state of the CI), so I would suggest merging this PR now and then addressing this issue in  a separate one.
finally, I think I would remove the unused `.travis.yml`, and any scripts that were used exclusively for the travis build, either in this PR or another one merged just after. Indeed, CI can seem a bit complex and I think having leftover files from builds that don't run anymore could add to confusion of maintainers or contributors

**Comment by NicolasGensollen** [comment]
@jeromedockes sounds good!

**Comment by jeromedockes** [comment]
great, thanks a lot @NicolasGensollen !

**Pull request merged**
